fmpz_mod_poly_factor/roots_general.c

```diff
@@ -0,0 +1,249 @@
+/*
+    Copyright (C) 2020 Daniel Schultz
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <http://www.gnu.org/licenses/>.
+*/
+
+#include "fmpz_mod_poly.h"
+#include "fmpz_mod_poly_factor.h"
+#include "fmpz.h"
+#include "fmpq.h"
+
+/* a = b */
+static void map_down(fmpz_mod_poly_t a, const fmpz_mod_poly_t b)
```

Comment:
What does this do?

Reply:
From it's implementation it looks like it reduced mod p. I don't think this should be valid. An fmpz_mod_poly should already have everything reduced mod p. The idea in fmpz_mod_poly is that unreduced coefficients would be in an fmpz_poly or a vector of fmpz's.

Reply:
a and b have different but compatible moduli, as the assert checks. I will make the names in this file better.